### PR TITLE
Edit enrollment.rb to cleanup and change lookup on app year

### PR DESCRIPTION
**Pull request by megcrenshaw**
Remove column_exists checks now that the migration has happened. See https://github.com/code-dot-org/code-dot-org/pull/44007.

Also modify `set_application_id` in `enrollments` to look for applications from the year of the enrollment's PD year, rather than from the current application year.

## Links

- jira ticket: [PLAT-1505](https://codedotorg.atlassian.net/browse/PLAT-1505)

## Testing story

Tests were added in https://github.com/code-dot-org/code-dot-org/pull/44021

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/config/foorm/forms/surveys/pd/summer_workshop_post_survey_test.0.json [hunk 1]
```diff
@@ -0,0 +1,238 @@
+{
```
Comment (bethanyaconnor) [inline dashboard/config/foorm/forms/surveys/pd/summer_workshop_post_survey_test.0.json]: Are these for tests?

File: dashboard/app/models/pd/enrollment.rb [hunk 2]
```diff
@@ -349,7 +333,7 @@ def set_application_id
     pd_match = ->(application) {application.try(:pd_workshop_id) == pd_workshop_id}
 
     application_id = nil
-    Pd::Application::ApplicationBase.where(user_id: user_id, application_year: APPLICATION_CURRENT_YEAR).each do |application|
+    Pd::Application::ApplicationBase.where(user_id: user_id, application_year: workshop.school_year).each do |application|
```
Comment (tim-dot-org) [inline dashboard/app/models/pd/enrollment.rb:338]: oo good idea! i think this works, but it also relies on the assumption that the workshops start after 6/1 (school_year assumes 6/1 is the start of the school year), might be worth a comment just incase it causes a problem later on

**Review by tim-dot-org — APPROVED** [review]

File: dashboard/config/foorm/forms/surveys/pd/summer_workshop_post_survey_test.0.json (hunk 1 above)
Comment (megcrenshaw) [reply dashboard/config/foorm/forms/surveys/pd/summer_workshop_post_survey_test.0.json]: Oh yikes! These aren't supposed to be in there ... just deleted ...

File: dashboard/app/models/pd/enrollment.rb (hunk 2 above)
Comment (megcrenshaw) [reply dashboard/app/models/pd/enrollment.rb:338]: Good call!

**Review by bethanyaconnor — APPROVED** [review]

**Pull request merged**